File: service/Controllers/ArtworkController.cs
using Microsoft.AspNetCore.Mvc;
using PaintLearning.Service.DTO;
using PaintLearning.Service.Services;

namespace PaintLearning.Service.Controllers;

[ApiController]
[Route("api/[controller]")]
public class ArtworkController : ControllerBase
{
    private readonly IAIService _aiService;
    private readonly IArtworkServiceClient _artworkServiceClient;

    public ArtworkController(IAIService aiService, IArtworkServiceClient artworkServiceClient)
    {
        _aiService = aiService;
        _artworkServiceClient = artworkServiceClient;
    }

    [HttpGet]
    public async Task<IActionResult> GetAllArtworks()
    {
        var result = await _artworkServiceClient.GetAllArtworksAsync();
        return Ok(result);
    }

    [HttpGet("{id}")]
    public async Task<IActionResult> GetArtworkById(string id)
    {
        var result = await _artworkServiceClient.GetArtworkByIdAsync(id);
        return Ok(result);
    }

    [HttpPost]
    public async Task<IActionResult> CreateArtwork([FromBody] ArtworkCreateRequest request)
    {
        var result = await _artworkServiceClient.CreateArtworkAsync(request);
        return Ok(result);
    }

    [HttpPut("{id}")]
    public async Task<IActionResult> UpdateArtwork(string id, [FromBody] ArtworkCreateRequest request)
    {
        var result = await _artworkServiceClient.UpdateArtworkAsync(id, request);
        return Ok(result);
    }

    [HttpDelete("{id}")]
    public async Task<IActionResult> DeleteArtwork(string id)
    {
        var result = await _artworkServiceClient.DeleteArtworkAsync(id);
        return Ok(result);
    }

    [HttpPost("{id}/favorite")]
    public async Task<IActionResult> ToggleFavorite(string id)
    {
        var result = await _artworkServiceClient.ToggleFavoriteAsync(id);
        return Ok(result);
    }

    [HttpPost("{id}/analyze")]
    public async Task<IActionResult> AnalyzeArtwork(string id, [FromBody] AnalysisRequest request)
    {
        var analysis = await _aiService.AnalyzeArtworkAsync(request.ImageBase64, request.Language);
        await _artworkServiceClient.AnalyzeArtworkAsync(id, request);
        return Ok(new ApiResponse<AnalysisResultDTO> { Code = 0, Message = "分析成功", Data = analysis });
    }

    [HttpPost("{id}/tutorial")]
    public async Task<IActionResult> GenerateTutorial(string id)
    {
        var artwork = await _artworkServiceClient.GetArtworkByIdAsync(id);
        if (artwork.Data?.Analysis == null)
        {
            return BadRequest(new ApiResponse<object> { Code = -1, Message = "请先分析画作" });
        }
        
        var tutorial = await _aiService.GenerateTutorialAsync(artwork.Data.Analysis);
        await _artworkServiceClient.GenerateTutorialAsync(id);
        return Ok(new ApiResponse<TutorialDTO> { Code = 0, Message = "教程生成成功", Data = tutorial });
    }

    [HttpPost("practice")]
    public async Task<IActionResult> CreatePractice([FromBody] PracticeRequest request)
    {
        var artwork = await _artworkServiceClient.GetArtworkByIdAsync(request.ArtworkId);
        if (artwork.Data?.ImageUri == null)
        {
            return BadRequest(new ApiResponse<object> { Code = -1, Message = "画作不存在" });
        }
        
        var feedback = await _aiService.ComparePracticeAsync(artwork.Data.ImageUri, request.PracticeImageBase64, request.Language);
        await _artworkServiceClient.CreatePracticeAsync(request);
        return Ok(new ApiResponse<FeedbackDTO> { Code = 0, Message = "练习反馈生成成功", Data = feedback });
    }
}

File: backend/src/main/java/com/paintlearning/app/controller/ArtworkController.java
package com.paintlearning.app.controller;

import com.paintlearning.app.dto.request.AnalysisRequest;
import com.paintlearning.app.dto.request.ArtworkCreateRequest;
import com.paintlearning.app.dto.request.PracticeRequest;
import com.paintlearning.app.dto.response.ApiResponse;
import com.paintlearning.app.dto.response.AnalysisResultDTO;
import com.paintlearning.app.dto.response.ArtworkDTO;
import com.paintlearning.app.dto.response.FeedbackDTO;
import com.paintlearning.app.dto.response.TutorialDTO;
import com.paintlearning.app.service.ArtworkService;
import jakarta.validation.Valid;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/artworks")
public class ArtworkController {

    @Autowired
    private ArtworkService artworkService;

    @GetMapping
    public ResponseEntity<ApiResponse<List<ArtworkDTO>>> getAllArtworks() {
        List<ArtworkDTO> artworks = artworkService.getAllArtworks();
        return ResponseEntity.ok(ApiResponse.success(artworks));
    }

    @GetMapping("/{id}")
    public ResponseEntity<ApiResponse<ArtworkDTO>> getArtworkById(@PathVariable String id) {
        ArtworkDTO artwork = artworkService.getArtworkById(id);
        return ResponseEntity.ok(ApiResponse.success(artwork));
    }

    @GetMapping("/favorites")
    public ResponseEntity<ApiResponse<List<ArtworkDTO>>> getFavoriteArtworks() {
        List<ArtworkDTO> favorites = artworkService.getFavoriteArtworks();
        return ResponseEntity.ok(ApiResponse.success(favorites));
    }

    @PostMapping
    public ResponseEntity<ApiResponse<ArtworkDTO>> createArtwork(@Valid @RequestBody ArtworkCreateRequest request) {
        ArtworkDTO artwork = artworkService.createArtwork(
                request.getImageUri(),
                request.getThumbnailUri(),
                request.getTags()
        );
        return ResponseEntity.ok(ApiResponse.success("创建成功", artwork));
    }

    @PutMapping("/{id}")
    public ResponseEntity<ApiResponse<ArtworkDTO>> updateArtwork(
            @PathVariable String id,
            @RequestBody ArtworkCreateRequest request) {
        ArtworkDTO artwork = artworkService.updateArtwork(
                id,
                request.getImageUri(),
                request.getThumbnailUri(),
                request.getTags()
        );
        return ResponseEntity.ok(ApiResponse.success("更新成功", artwork));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<ApiResponse<Void>> deleteArtwork(@PathVariable String id) {
        artworkService.deleteArtwork(id);
        return ResponseEntity.ok(ApiResponse.success("删除成功", null));
    }

    @PostMapping("/{id}/favorite")
    public ResponseEntity<ApiResponse<ArtworkDTO>> toggleFavorite(@PathVariable String id) {
        ArtworkDTO artwork = artworkService.toggleFavorite(id);
        return ResponseEntity.ok(ApiResponse.success(artwork));
    }

    @PostMapping("/{id}/analyze")
    public ResponseEntity<ApiResponse<AnalysisResultDTO>> analyzeArtwork(
            @PathVariable String id,
            @Valid @RequestBody AnalysisRequest request) {
        AnalysisResultDTO analysis = artworkService.analyzeArtwork(id, request);
        return ResponseEntity.ok(ApiResponse.success("分析成功", analysis));
    }

    @PostMapping("/{id}/tutorial")
    public ResponseEntity<ApiResponse<TutorialDTO>> generateTutorial(@PathVariable String id) {
        TutorialDTO tutorial = artworkService.generateTutorial(id);
        return ResponseEntity.ok(ApiResponse.success("教程生成成功", tutorial));
    }

    @PostMapping("/practice")
    public ResponseEntity<ApiResponse<FeedbackDTO>> createPractice(@Valid @RequestBody PracticeRequest request) {
        FeedbackDTO feedback = artworkService.createPractice(request);
        return ResponseEntity.ok(ApiResponse.success("练习反馈生成成功", feedback));
    }
}